gson/src/main/java/com/google/gson/internal/Excluder.java

```diff
@@ -199,7 +199,8 @@ private boolean excludeClassChecks(Class<?> clazz) {
           return true;
       }
 
-      if (isAnonymousOrLocal(clazz)) {
+      // Permit static local classes, such as Records (Java 16 feature)
+      if (!isStatic(clazz) && isAnonymousOrLocal(clazz)) {
```

Comment:
I am wondering if it makes sense for `isAnonymousOrLocal` to become `isAnonymousOrLocalAndNotStatic`. In the other case where we call the method (`excludeField`), would we want to accept static local classes as well?

Reply:
Ah, you are right `excludeField` should not exclude fields with static local classes as type either.
Moving it into `isAnonymousOrLocal` sounds good. What about `isAnonymousOrNonStaticLocal` as name instead?

Reply:
Sure, sounds good.

Reply:
Have force-pushed the change. It now in theory also allows static anonymous classes (to not make the boolean expression too nested), but I don't think such a thing is possible in Java at the moment.